pkg/util/metric/metric.go

```diff
@@ -467,9 +469,59 @@ func (h *Histogram) CumulativeSnapshot() HistogramSnapshot {
 	return MakeHistogramSnapshot(h.ToPrometheusMetric().Histogram)
 }
 
+// collects various private fields from a prometheus histogram to debug
+// whether the lengths of various internal slices are mis-matched.
+func collectHistogramBoundsCountsLengths(h prometheus.HistogramInternal) (resp string) {
+	// handle panics in case we change the histogram version
+	defer func() {
+		if r := recover(); r != nil {
+			resp = fmt.Sprintf("%v", r)
+		}
+	}()
+	v := reflect.ValueOf(h)
+	for v.Kind() == reflect.Interface || v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+	upperBounds := v.FieldByName("upperBounds")
+	parts := []string{fmt.Sprintf("len(upperBounds): %d", upperBounds.Len())}
+	counts := v.FieldByName("counts")
+	for i := 0; i < counts.Len(); i++ {
+		ptr := counts.Index(i)
+		if ptr.IsNil() {
+			// If the pointer is nil, just report that
+			parts = append(parts, fmt.Sprintf("len(counts[%d].buckets): nil", i))
+			continue
+		}
+		// Dereference to get the histogramCounts struct
+		countStruct := ptr.Elem()
+
+		// Within the struct, get the "buckets" field (a slice)
+		bucketsField := countStruct.FieldByName("buckets")
+		parts = append(parts, fmt.Sprintf("len(counts[%d].buckets): %d", i, bucketsField.Len()))
+	}
+	resp = strings.Join(parts, ", ")
+	return resp
+}
+
+// DebugAugmentPanic attaches information about a panicking goroutine from
+// a seemingly impossible situation seen in the ticket
+// https://github.com/cockroachdb/cockroach/issues/136186.
+// The goal is to see how the values of h.windowed.[cur/prev].[upperBounds/counts]
+// differ.
+func (h *Histogram) debugAugmentPanic() {
+	if r := recover(); r != nil {
+		stack := debug.Stack()
```

Comment:
Might want to consider using this instead: https://github.com/cockroachdb/cockroach/blob/1eceee107d663e1fa16db4ea9541f0806a4644b3/pkg/util/debugutil/debugutil.go#L62

Not sure how this panic will get propagated to us (logs? sentry?). If it's through logs, it might get redacted.